### PR TITLE
[CES-19] Azure DevOps Agent infra moved to core modules

### PR DESCRIPTION
**Pull request by christian-calabrese**
### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [ ] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/core/_modules/azure_devops_agent/main.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  private_endpoint_network_policies_enabled = false
+
+  service_endpoints = [
+    "Microsoft.Web",
+  ]
+}
+
+module "azdoa_li_infra" {
+  source              = "github.com/pagopa/terraform-azurerm-v3//azure_devops_agent?ref=v8.27.0"
+  name                = try(local.nonstandard[var.location_short].li_infra, "${var.project}-azdoa-vmss-li-infra-01")
```
Comment (Krusty93) [inline src/core/_modules/azure_devops_agent/main.tf:24]:
```suggestion
  name                = try(local.nonstandard[var.location_short].li_infra, "${var.project}-azdoa-infra-vmss-01")
```

File: src/core/app_backend.tf [hunk 2]
```diff
@@ -717,7 +717,7 @@ module "appservice_app_backendl1_slot_staging" {
   ip_restriction_default_action = "Deny"
 
   allowed_subnets = [
-    module.azdoa_snet[0].id,
+    data.azurerm_subnet.azdoa_snet.id,
```
Comment (gunzip) [inline src/core/app_backend.tf:720]: do we still need to let the devops agent access io-backend?

File: src/core/_modules/azure_devops_agent/main.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  location            = var.location
+  source_image_name   = var.image_name
+  vm_sku              = "Standard_B2als_v2"
+
+  tags = var.tags
+}
+
+module "azdoa_loadtest_li" {
+  source              = "github.com/pagopa/terraform-azurerm-v3//azure_devops_agent?ref=v8.27.0"
+  name                = try(local.nonstandard[var.location_short].li_loadtest, "${var.project}-azdoa-vmss-li-loadtest-01")
```
Comment (Krusty93) [inline src/core/_modules/azure_devops_agent/main.tf:37]:
```suggestion
  name                = try(local.nonstandard[var.location_short].li_loadtest, "${var.project}-azdoa-loadtest-vmss-01")
```

File: src/core/prod/westeurope.tf [hunk 4]
```diff
@@ -108,3 +108,23 @@ module "vpn_weu" {
...
+  resource_group_name = data.azurerm_resource_group.common_weu.name
+  project             = local.project_weu_legacy
+
+  vnet_common     = module.networking_weu.vnet_common
+  resource_groups = local.resource_groups[local.location_short[data.azurerm_resource_group.common_weu.location]]
+  datasources = {
+    azurerm_client_config = data.azurerm_client_config.current
+  }
+
+  image_name  = "azdo-agent-ubuntu2204-image-v2"
```
Comment (Krusty93) [inline src/core/prod/westeurope.tf:126]: Small optional suggestion, I'd move this value inside the module

**Review by Krusty93 — APPROVED** [review]

File: src/core/app_backend.tf (hunk 2 above)
Comment (christian-calabrese) [reply src/core/app_backend.tf:720]: It looks like io-backend is still deployed via Azure DevOps and no private endpoint is present on the machines. I think we should keep this rule

**Pull request merged**